cmd/podman/images/scp.go

```diff
@@ -0,0 +1,354 @@
+	if err := parse.ValidateFileName(saveOpts.Output); err != nil {
+		return err
+	}
+	confR, err := config.NewConfig("") // create a hand made config for the remote engine since we might use remote and native at once
+	if err != nil {
+		return errors.Wrapf(err, "could not make config")
+	}
+	abiEng, err := registry.NewImageEngine(cmd, args) // abi native engine
+	if err != nil {
+		return err
+	}
+
+	cfg, err := config.ReadCustomConfig() // get ready to set ssh destination if necessary
+	if err != nil {
+		return err
+	}
+	serv, err := parseArgs(args, cfg) // parses connection data and "which way" we are loading and saving
+	if err != nil {
+		return err
+	}
+	// TODO: Add podman remote support
+	confR.Engine = config.EngineConfig{Remote: true, CgroupManager: "cgroupfs", ServiceDestinations: serv} // pass the service dest (either remote or something else) to engine
+	switch {
+	case scpOpts.FromRemote: // if we want to load FROM the remote
+		err = savefromRemote(scpOpts.SourceImageName, scpOpts.Save.Output, "", scpOpts.URI[0], scpOpts.Iden[0])
+		if err != nil {
+			return err
+		}
+		if scpOpts.ToRemote { // we want to load remote -> remote
+			rep, err := savetoRemote(scpOpts.Save.Output, "", scpOpts.URI[1], scpOpts.Iden[1])
+			if err != nil {
+				return err
+			}
+			fmt.Println(rep)
+			break
+		}
+		report, err := abiEng.Load(context.Background(), scpOpts.Load)
+		if err != nil {
+			return err
+		}
+		fmt.Println("Loaded images(s): " + strings.Join(report.Names, ","))
+	case scpOpts.ToRemote: // remote host load
+		scpOpts.Save.Format = "oci-archive"
+		abiErr := abiEng.Save(context.Background(), scpOpts.SourceImageName, []string{}, scpOpts.Save) // save the image locally before loading it on remote, local, or ssh
+		if abiErr != nil {
+			errors.Wrapf(abiErr, "could not save image as specified")
+		}
+		rep, err := savetoRemote(scpOpts.Save.Output, "", scpOpts.URI[0], scpOpts.Iden[0])
+		if err != nil {
+			return err
+		}
+		fmt.Println(rep)
+	// TODO: Add podman remote support
+	default: // else native load
+		if scpOpts.Tag != "" {
+			return errors.Wrapf(define.ErrInvalidArg, "Renaming of an image is currently not supported")
+		}
+		scpOpts.Save.Format = "oci-archive"
+		abiErr := abiEng.Save(context.Background(), scpOpts.SourceImageName, []string{}, scpOpts.Save) // save the image locally before loading it on remote, local, or ssh
+		if abiErr != nil {
+			errors.Wrapf(abiErr, "could not save image as specified")
+		}
+		rep, err := abiEng.Load(context.Background(), scpOpts.Load)
+		if err != nil {
+			return err
+		}
+		fmt.Println("Loaded images(s): " + strings.Join(rep.Names, ","))
+	}
+	return nil
+}
+
+// toRemote takes a dialed ssh client, and image information. it connects to the client
+// and copies the saved image dir over to the remote host and then loads it onto the machine
+// returns a string containing output or an error
+func savetoRemote(localFile string, tag string, url *urlP.URL, iden string) (string, error) {
```

Comment:
Shouldn’t this be `loadFromRemote`, as a counterpart to `saveFromRemote`? I could see a case for naming this one “save”, but naming _both_ directions “save” is weird.

Also in the local case the code calls `abiEng.Load`, so having the remote case named similarly would be easier.

Reply:
I think `loadToRemote` and `saveToRemote` is more accurate since the final command is remote load in `loadToRemote` and remote save in `saveToRemote` changed it to that.